### ISSUE TITLE
phydro issues

### ISSUE THREAD

**Issue by stineb** [geco-bern/rsofun#218]
- [ ] Resolve compilation error on M1 Mac (may test on a different platform with rhub checks, see below):
```
  dlopen(/Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library/00LOCK-rsofun/00new/rsofun/libs/rsofun.so, 0x0006): symbol not found in flat namespace '___gcc_nested_func_ptr_created'
Error: loading failed
Execution halted
ERROR: loading failed
```
- [ ] Run `devtools::check()` and resolve all issues.
- [ ] Run rhub checks, see [here](https://blog.r-hub.io/2024/04/11/rhub2/). 
- [ ] vignettes folder must contain only `.Rmd` files. Scripts go into `analysis/`. Function definitions go into `R/`.
- [ ] Build website locally (`pkgdown::build_site()`) and clean it up - All `.Rmd` files in `vignettes/` will appear as 'Articles'. This will be public-facing. Make it nice.
- [ ] Make sure that all FluxDataKit-related code is _not_ in {rsofun}.

**Comment by stineb** [comment]
Points resolved (not yet all) as per PR https://github.com/geco-bern/rsofun/pull/219

**Comment by stineb** [comment]
Build of `sensititivity_analysis.Rmd` now fails with error:
```r
> morrisOut.df <- data.frame(
+   parameter = names(par_cal_best),
+   mu.star = apply(abs(morrisOut$ee), 2, mean, na.rm = T),
+   sigma = apply(morrisOut$ee, 2, sd, na.rm = T)
+ ) %>%
+   arrange( mu.star )
Error in data.frame(parameter = names(par_cal_best), mu.star = apply(abs(morrisOut$ee),  : 
  arguments imply differing number of rows: 11, 10
```
Should run this again but if fails, too:
```r
> morrisOut <- sensitivity::morris(
+   model = morris_setup$posterior$density,
+   factors = names(par_cal_best), 
+   r = 1000, 
+   design = list(type = "oat", levels = 20, grid.jump = 3), 
+   binf = par_cal_min, 
+   bsup = par_cal_max, 
+   scale = TRUE)
Warning in dunif(x, min = lower, max = upper, log = T) : NaNs produced
Error in if (out == Inf) stop("Inf encountered in prior") : 
  missing value where TRUE/FALSE needed
```
Has number of model parameter or their names changed with the PR?

**Comment by stineb** [comment]
Can PM be used in combination with P-model-Gs? There is code that is now commented but may have to be un-commented and completed?
In `gpp_pmodel.mod.f90`:
```Fortran
      if (use_pml) then
        ! We plug Pmodel/Phydro-derived gs into the PM equation to calculate T (note this is uncoupled PM-transpiration)
        ! use PFT-specific gs for this calculation: tile_fluxes(lu)%plant(pft)%gs_accl

        ! TODO: Fill this in
        ! tile_fluxes(lu)%plant(pft)%dtransp = PM_EQUATION(tile_fluxes(lu)%plant(pft)%gs_accl)
```
The argument `using_pml` is not used in subroutines of `waterbal_splash.mod.f90`.
@jaideep777

**Comment by jaideep777** [comment]
Yes, as mentioned in the comment, this is just a placeholder where the PM equation should go. It's not coded in yet.

**Comment by stineb** [comment]
I think the PM should be applied at the LU-level (land unit-level), not PFT-level. I suggest to use the fractional plant coverage-weighted average Gs and calculate PML-ET in the waterbal module (it's fine to duplicate code that is already in `photosynth_phydro.mod.f90` for now).

**Comment by jaideep777** [comment]
Isn't the whole point of PM to distinguish surface properties of different vegetation types? If you want to do this only for inter-cell differences, then yes, the entire code chunk with the T calculation can be moved to waterbal_splash

**Comment by stineb** [comment]
The gridcell-average (or land unit-average) surface properties can be used. The model structure is adopted from LPX (and other similar models) where all PFTs have access to the same soil water and the soil water balance is done at the gridcell-level (or land-unit level).
I see that there will be an inconsistency when transpiration is calculated per PFT following diffusion, and then total canopy-transpiration is calculated at the gridcell-level follwoing PML. They won't add up.
